frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/utils.ts

```diff
@@ -304,12 +310,19 @@ export const getTopologyNodesEdges = (
   return { nodes, edges };
 };
 
+export const hasWhenExpression = (pipeline: PipelineKind): boolean => {
+  return [...(pipeline?.spec?.tasks ?? []), ...(pipeline?.spec?.finally ?? [])].some(
```

Comment:
fwiw, `||` is probably more applicable here since `tasks` and `finally` are only ever arrays. If you could have `''` or `0` or `false` as valid values, it would make sense to do `??` but if you're either "not defined" (null / undefined) or an array, it's easier to just do `||` (imo).

Entirely a nit, arguably just a preference, and definitely not important.